shotover-proxy/src/message/mod.rs

```diff
@@ -307,33 +315,53 @@ pub enum QueryType {
     PubSubMessage,
 }
 
-#[derive(PartialEq, Debug, Clone, Serialize, Deserialize)]
+#[derive(PartialEq, Debug, Clone, Serialize, Deserialize, Hash, Eq, PartialOrd, Ord)]
 pub enum Value {
     NULL,
     None,
     #[serde(with = "my_bytes")]
     Bytes(Bytes),
+    Ascii(String),
     Strings(String),
-    Integer(i64),
-    Float(f64),
+    Integer(i64, IntSize),
+    Double(OrderedFloat<f64>),
+    Float(OrderedFloat<f32>),
     Boolean(bool),
     Inet(IpAddr),
     List(Vec<Value>),
     Rows(Vec<Vec<Value>>),
-    NamedRows(Vec<HashMap<String, Value>>),
-    Document(HashMap<String, Value>),
+    NamedRows(Vec<BTreeMap<String, Value>>),
+    Document(BTreeMap<String, Value>),
     FragmentedResponse(Vec<Value>),
+    Set(BTreeSet<Value>),
+    Map(BTreeMap<Value, Value>),
+    Varint(BigInt),
+    Decimal(BigDecimal),
+    Date(i32),
+    Timestamp(i64),
+    Timeuuid(Uuid),
+    Varchar(String),
+    Uuid(Uuid),
+    Time(i64),
+    Counter(i64),
+    Tuple(Vec<Value>),
+    Udt(BTreeMap<String, Value>),
+}
+
+#[derive(PartialEq, Debug, Clone, Serialize, Deserialize, Hash, Eq, PartialOrd, Ord)]
+pub enum IntSize {
+    I64, // BigInt
+    I32, // Int
+    I16, // Smallint
+    I8,  // Tinyint
 }
 
 impl From<Frame> for Value {
     fn from(f: Frame) -> Self {
-        // panic!("Aug 07 15:56:49.621  INFO instaproxy::transforms::tuneable_consistency_scatter: Response(QueryResponse { matching_query: None, original: Redis(BulkString([102, 111, 111])), result: Some(Strings("foo")), error: None })
-        // ");
-        //         panic!("This should be a bulk_string(byte array) and not be a string"); // I wonder if the bytes themselves need to get serialised differently....?
         match f {
             Frame::SimpleString(s) => Value::Strings(s),
             Frame::Error(e) => Value::Strings(e),
-            Frame::Integer(i) => Value::Integer(i),
+            Frame::Integer(i) => Value::Integer(i, IntSize::I32),
```

Comment:
This should be IntSize::I64 as per https://redis.io/topics/protocol#resp-integers